### ISSUE TITLE
No warnings when running under `pre-commit`

### ISSUE THREAD

**Issue by GarrisonD** [ansible/ansible-lint#2855]
##### Summary

Some rules (e.g. `no-free-form`, `role-name`, `name[play]`) are not reported when running under `pre-commit`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 6.10.1 using ansible 2.14.1
```

- `ansible` installation method: no `ansible`, only `ansible-core`
- `ansible-lint` installation method: pip, pre-commit

##### STEPS TO REPRODUCE

```
$ docker run --rm -it python bash

/# apt update && apt install nano

/# cd home

/home# nano test.yml

---
- hosts: all

  tasks:
    - name: Update APT cache
      ansible.builtin.package: update_cache=true cache_valid_time=3600

/home # nano .pre-commit-config.yaml

repos:
  - repo: https://github.com/ansible/ansible-lint
    rev: v6.10.1
    hooks:
      - id: ansible-lint

/home# pip install ansible-lint pre-commit

/home# ansible-lint test.yml

WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: test.yml
WARNING  Listing 2 violation(s) that are fatal
name[play]: All plays should be named. (warning)
test.yml:2

no-free-form: Avoid using free-form when calling module actions. (ansible.builtin.package) (warning)
test.yml:5 Task/Handler: Update APT cache

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

                      Rule Violation Summary                      
 count tag          profile  rule associated tags                 
     1 name[play]   basic    idiom (warning)                      
     1 no-free-form moderate syntax, risk, experimental (warning) 

Passed with min profile: 0 failure(s), 2 warning(s) on 1 files.

/home# git init

/home# pre-commit run -v

Ansible-lint.............................................................Passed
- hook id: ansible-lint
- duration: 1.73s

INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/4ea140/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/4ea140/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/4ea140/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Discovered files to lint using: git ls-files --cached --others --exclude-standard -z
INFO     Excluded removed files using: git ls-files --deleted -z

Passed with production profile: 0 failure(s), 0 warning(s) on 2 files.
```

##### Desired Behavior

Warnings are collected and reported.

##### Actual Behavior

No warnings are collected and reported.

`test.yml`

```yml
---
- hosts: all

  tasks:
    - name: Update APT cache
      ansible.builtin.package: update_cache=true cache_valid_time=3600
```

`.pre-commit-config.yaml`

```yml
repos:
  - repo: https://github.com/ansible/ansible-lint
    rev: v6.10.1
    hooks:
      - id: ansible-lint
```

**Comment by GarrisonD** [comment]
I made a mistake: the equivalent command for `pre-commit run -v` is `ansible-lint -v` (not `ansible-lint test.yml`).

And now it's a question of how to make `ansible-lint` treating `playbook.yml` sitting in the root dir as a playbook instead of just some YAML file.

**Comment by GarrisonD** [comment]
I narrowed down the issue even more.

So the `playbook.yml` sitting in the root dir really has the `playbook` kind, but I can see failures/warnings only when I comment out this line: `- role: newrelic.newrelic-infra`.

Logs of `ansible-lint -vvvv`:

```console
...

INFO     Executing syntax check on playbook.yml (4.31s)
DEBUG    Skipping /home/vscode/.cache/ansible-compat/cd0c3e/roles/newrelic.newrelic-infra/tasks/install_dist_pkgs.yml as it is outside of the project directory.
DEBUG    Attempting to release lock 139955563604752 on /home/vscode/.cache/ansible-compat/cd0c3e/.lock
DEBUG    Lock 139955563604752 released on /home/vscode/.cache/ansible-compat/cd0c3e/.lock

...
```

I have several other roles used as well but only `newrelic.newrelic-infra` is here 🤔

**Comment by GarrisonD** [comment]
I tried this:

```yml
---
mock_roles:
  - newrelic.newrelic-infra
```

but didn't help.

**Comment by ssbarnea** [comment]
I think that you mislead how pre-commit works. It is no concept of warnings, only pass or fail and when it passes it hides all the output generated by the passed hook.

This means that you need to make it run in strict mode if you want to be able to act on warnings... basically strict converts all warnings to errors. If you don't like this, I invite you to ask on https://github.com/pre-commit/pre-commit/

Also, please use discussions instead of issue tracker.
